docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+            cur.execute(query)
+
+Training A Model
+----------------
+
+First, we will load the data into our python context with ``Psycopg2`` and
+``Pandas``
+
+.. code-block:: python
+
+    import pandas.io.aql as sqlio
+    from psycopg2 import connect
+
+
+    with connect(conn_str) as conn:
+        query = "SELECT * FROM ml.pump_sensors;"
+        df = sqlio.read_sql_query(query, conn)
+
+The data was successfully loaded for the application and we can now start
+to prepare the data so we can train a `Multilayer Perceptron`_ (MLP) to
+classify the machine-status in a given time window.
+
+We will make an assumption and combine the two labels ``RECOVERING``
+and ``BROKEN`` and combine them as a single label, so we can either say
+the machine is in the state ``NORMAL`` or ``BROKEN``. Then we will binarize
+the labels so we can use them in the MLP. Furthermore we need to extract
+the sensor columns and split the data into train and test data frames while
+normalizing their values between (0,1).
+
+.. code-block:: python
+
+    from sklearn.preprocessing import MinMaxScaler, label_binarize
+
+
+    df["machine_status"] = df["machine_status"].replace("RECOVERING", "BROKEN")
+    df["machine_status"] = label_binarize(df["machine_status"], classes=["BROKEN", "NORMAL"])
+
+    min_max_scaler = MinMaxScaler(feature_range=(0, 1))
+
+    test_split = 70
+    df_size = len(df.index)
+    split = int((df_size / 100) * test_split)
+
+    df_train = df.iloc[0:split, :]
+    train_data_cols = df_train.iloc[:, 2:54]
+    norm_train_data_cols = min_max_scaler.fit_transform(train_data_cols)
+
+    df_test = df.iloc[split:len(df.index), :]
+    test_data_cols = df_test.iloc[:, 2:54]
+    norm_test_data_cols = min_max_scaler.transform(test_data_cols)
+
+The next step would be to define our time steps for the data. Let's say we take
+an input time of 60 minutes and try to predict if the machine will fail in the next
+12 hours or 720 minutes. After defining these we will create data pairs for training
+the MLP model that fit into the time steps.
+
+.. code-block:: python
+
+    # Add this function over the main function.
+    def add_labels(start, stop, dataset):
+        for step in range(start, stop):
+            if dataset["machine_status"].iloc[step] == 0:
+                return 0
+        return 1
+
+
+    timestep = 60
+    output_timestep = 720
+
+    train_pairs = []
+
+    saved_step = 0
+    for i in df_train.index:
+        row = saved_step
+        if row + timestep + output_timestep < len(df_train.index):
+            data_window = norm_train_data_cols[row:row + timestep]
+            data_vector = data_window.flatten()
+            state = add_labels(row + timestep, row + timestep + output_timestep, df_train)
+            train_pairs.append((data_vector, state))
+            saved_step = row + timestep
+        else:
+            break
+
+Now that we have the data pairs we can start creating our x/y training data frames.
```

Comment:
Could you briefly explain what this means? Perhaps say to train the data, we need to split it into 2 sets: x and y. Set x will be used to train the model, and set y will be used to verify the model's accuracy. (Assuming that I understand this correctly :sweat_smile:)